### PR TITLE
test: optimize k8s test

### PR DESCRIPTION
**Pull request by bwanglzu**
Currently, for every test, we always load all docker images into kind k8s cluster. In this PR, we only load **needed** container image w.r.t the test case. For some tests, we add a jina pod without using container image, this loading has been completely removed.

As @florian-hoenicke suggested, we could further pre-build docker images/create cluster for only one time. Not included in this PR.

master: 1296.60s, feature branch: 594.21s (`test_k8s.py`, local machine).

DO NOT MERGE, we'll incorporate this change to star routing branch.

### CODE REVIEW

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by bwanglzu** [comment]
do not merge until we have fixed grpc issue.

**Review by jacobowitz — COMMENTED** [review]
very nice PR! I have no clue why the connection pool test is acting up, maybe we can disable it if its not easy to fix? We are reworking this atm anyway on the star routing branch.

File: tests/k8s/test_graceful_request_handling.py [hunk 1]
```diff
@@ -165,21 +162,19 @@ async def test_no_message_lost_during_scaling(
...
-async def test_no_message_lost_during_kill(
-    slow_process_executor_image,
-    logger,
-    k8s_cluster,
-    load_images_in_kind,
-    set_test_pip_version,
-):
+async def test_no_message_lost_during_kill(logger, k8s_cluster, image_name_tag_map):
+    image_names = ['slow-process-executor']
+    images = [
```
[the remark this answers is not in the document]
Comment (bwanglzu) [reply tests/k8s/test_graceful_request_handling.py:167]: yes we can, i also want to do it (now I have more time)

**Comment by jacobowitz** [comment]
I've incorporated this PR into star routing here https://github.com/jina-ai/jina/pull/3964
I think its a great speed up!
Will close this PR in favor of the other one

**Pull request closed without merging**